homeassistant/components/freedompro/climate.py

```diff
@@ -52,7 +57,7 @@ async def async_setup_entry(
 class Device(CoordinatorEntity, ClimateEntity):
     """Representation of an Freedompro climate."""
 
-    _attr_hvac_modes = SUPPORTED_HVAC_MODES
+    _attr_hvac_modes = SUPPORTED_HVAC_MODES  # type:ignore[assignment]
```

Comment:
Why is this needed?

Reply:
In the platform it is defined as `list[HVACMode | str]`, and mypy complains.
https://github.com/home-assistant/core/blob/8db499434567a72012d9dd3e21d435f28e82941a/homeassistant/components/climate/__init__.py#L194

My understanding is that at the plaform level `str` was only kept temporarily for compatibility.